sources/packages/backend/apps/api/src/route-controllers/application-change-request/_tests_/e2e/application-change-request.aest.assessApplicationChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,482 @@
+      {
+        id: expect.any(Number),
+        description: payload.note,
+        creator: ministryUser,
+      },
+    ]);
+  });
+
+  it("Should approve a change request and copy the offering and no appeals when the application change request is waiting for approval, and no appeals are present.", async () => {
+    // Arrange
+    // Application to be replaced by the change request.
+    const applicationToBeReplaced = await saveFakeApplication(
+      appDataSource,
+      undefined,
+      {
+        initialValues: {
+          applicationStatus: ApplicationStatus.Completed,
+          applicationEditStatus: ApplicationEditStatus.Original,
+        },
+      },
+    );
+    // Change request to be approved replacing the "applicationToBeReplaced".
+    const changeRequest = await saveFakeApplication(
+      appDataSource,
+      { precedingApplication: applicationToBeReplaced },
+      {
+        initialValues: {
+          applicationStatus: ApplicationStatus.Edited,
+          applicationEditStatus: ApplicationEditStatus.ChangePendingApproval,
+        },
+      },
+    );
+    // Offering to perform the asserts. Expected to be copied.
+    const expectedOffering = {
+      id: applicationToBeReplaced.currentAssessment.offering.id,
+    };
+    // Appeal to perform the asserts. Expected to be copied.
+    const payload = getPayload(ApplicationEditStatus.ChangedWithApproval);
+    const endpoint = `/aest/application-change-request/${changeRequest.id}`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+    const now = new Date();
+    MockDate.set(now);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send(payload)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+    // Validate database changes (focus on offering and appeal validation only).
+    const applicationChangeRequest = await db.application.findOne({
+      select: {
+        id: true,
+        currentAssessment: {
+          id: true,
+          offering: {
+            id: true,
+          },
+          studentAppeal: {
+            id: true,
+          },
+        },
+        precedingApplication: {
+          id: true,
+          currentAssessment: {
+            id: true,
+            offering: {
+              id: true,
+            },
+            studentAppeal: {
+              id: true,
+            },
+          },
+        },
+      },
+      relations: {
+        currentAssessment: {
+          offering: true,
+          studentAppeal: true,
+        },
+        precedingApplication: {
+          currentAssessment: {
+            offering: true,
+            studentAppeal: true,
+          },
+        },
+      },
+      where: {
+        id: changeRequest.id,
+      },
+      loadEagerRelations: false,
+    });
+    // Focus on offering and appeal validation only.
+    expect(applicationChangeRequest).toEqual(
+      expect.objectContaining({
```

Comment:
Is `objectContaining`  required as we validate only what we select.

Reply:
It is a leftover from a previous change, I will remove it.